sdk/cosmos/azure-cosmos/azure/cosmos/aio/_user.py

```diff
@@ -118,12 +124,15 @@ def list_permissions(self, **kwargs: Any) -> AsyncItemPaged[Dict[str, Any]]:
     @distributed_trace
     def query_permissions(
         self,
-        query: Union[str, Dict[str, Any]],
```

Comment:
I'm not sure why - but throughout the async modules, `query` was typed as `Union[str, Dict]` - was this intentional based on a customer request? I couldn't really see a reason for it, and it was broken if both `query` and `parameters` were specified.
Note that I haven't actually changed any behaviour here, just the type hint. I can revert this change throughout if that would be preferable.

Reply:
good catch, I am honestly not sure why the typing was set that way to begin with - maybe it was supported at some point, but from the code logic like you said it would not have worked anyway: 
<img width="486" alt="image" src="https://github.com/Azure/azure-sdk-for-python/assets/30335873/0a08321b-52a6-4062-a48d-cc31717c0d93">